iceoryx_hoofs/vocabulary/include/iox/string.hpp

```diff
@@ -534,7 +534,7 @@ class string final
 
     // safe access is guaranteed since the char array is wrapped inside the string class
     // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
-    char m_rawstring[Capacity + 1]{};
+    char m_rawstring[Capacity + 1]{'\0'};
```

Comment:
Oh I know, this line was once before in this state, then we removed the `\0` and now we introduce it again. One wild thought here: Shall we maybe disable the array bounds checking in gcc. I have the impression that it causes over and over false positives, where we adjust the code just to make the gcc version happy and the next gcc version struggles again.

Reply:
This is not strictly related to the bounds checks. You can call `c_str` on an default constructed string and get a non null-terminated garbage. At least if I'm not totally mistaken.